core/arch/arm/kernel/secure_partition.c

```diff
@@ -1645,11 +1612,6 @@ static TEE_Result sp_init_all(void)
 				panic();
 		}
 	}
-	/*
-	 * At this point all FIP SPs are loaded by ldelf so the original images
-	 * (loaded by BL2 earlier) can be unmapped
-	 */
-	fip_sp_unmap_all();
```

Comment:
IIUC moving this call up (before the the `sp_first_run()` calls) is unrelated to this fix, in which case it would be better to create a separate commit.

Reply:
Okay, I'll move this to a separate commit.